sincere-singularities/.pre-commit-config.yaml

```
@@ -0,0 +1,29 @@
+# Pre-commit configuration.
+# See https://github.com/python-discord/code-jam-template/tree/main#pre-commit-run-linting-before-committing
+
+repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.6.0
+    hooks:
+      - id: check-added-large-files
+      - id: check-ast
+      - id: check-case-conflict
+      - id: check-json
+      - id: check-toml
+      - id: check-yaml
+      - id: end-of-file-fixer
+      - id: trailing-whitespace
+        args: [--markdown-linebreak-ext=md]
+
+  - repo: https://github.com/astral-sh/ruff-pre-commit
+    rev: v0.5.2
+    hooks:
+      - id: ruff
+        args: [--fix]
+      - id: ruff-format
+
+  - repo: https://github.com/pre-commit/mirrors-mypy
+    rev: v1.10.1
+    hooks:
+      - id: mypy
+        # args: [--strict]  # Put config in pyproject.toml. If it doesn't work, put it here.
```

Putting your requirements in `additional_dependencies` will allow mypy to type check usage of imported code.
Looks like it wasn't able to catch some issues in `bot.py` because it couldn't find the dependencies.
https://github.com/pre-commit/mirrors-mypy?tab=readme-ov-file#using-mypy-with-pre-commit

```suggestion
  - repo: https://github.com/pre-commit/mirrors-mypy
    rev: v1.10.1
    hooks:
      - id: mypy
        additional_dependencies:
          - disnake~=2.9.2
          - python-dotenv~=1.0.1
          - dacite~=1.8.1
          - pymongo~=4.8.0
          - torch~=2.3.1
          - sentence-transformers~=3.0.1
          - transformers~=4.43.2
          - numpy<2
          - faker~=26.0.0
```